### ISSUE TITLE
Error collect2: fatal error: cannot find 'ld' when debugging

### ISSUE THREAD

**Issue by allen505** [microsoft/vscode-cpptools#5800]
Type: Debugger
I am trying to run a simple `Hello World` program(the issue pertains to other programs I tried too). I get the following error, when I tried to run the debugger
```
Executing task: /usr/bin/g++ -g /home/.../test.cpp -o /home/.../test <

collect2: fatal error: cannot find 'ld'
compilation terminated.
The terminal process terminated with exit code: 1
```
I tried running the command
`/usr/bin/g++ -g /home/.../test.cpp -o /home/.../test` itself and it compiled successfully.

**I remember using the debugger a couple of weeks ago and it used to work fine. I haven't had any system updates since and don't remember making an changes to any packages.**

**Describe the bug**
- OS and Version: Manjaro Lysia 20.0.3
- VS Code Version: 1.45.1
- C/C++ Extension Version: 0.29.0
- Other extensions you installed : Only Atom One Dark Theme, which shouldn't cause an issue
- A clear and concise description of what the bug is:
```sh
❯ which ld
/bin/ld

❯ ld --version
GNU ld (GNU Binutils) 2.34.0

❯ /usr/bin/ld --version
GNU ld (GNU Binutils) 2.34.0

❯ g++ --version
g++ (GCC) 10.1.0
```
Unable to use the debugger because of the error stated above. Issue pertains to a variety of the programs I tried it on.

**To Reproduce**
Steps to reproduce the behaviour:
1. Write any C code
2. Run the debugger
3. Debugging fails with error state above

**Additional context**
Sample Code:
```c
#include <iostream>
using namespace std;

int main()
{
    cout<<"Hello World";
}
```
launch.json:
```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "g++ - Build and debug active file",
            "type": "cppdbg",
            "request": "launch",
            "program": "${fileDirname}/${fileBasenameNoExtension}",
            "args": ["lm","W"],
            "stopAtEntry": false,
            "cwd": "${workspaceFolder}",
            "environment": [],
            "externalConsole": false,
            "MIMode": "gdb",
            "setupCommands": [
                {
                    "description": "Enable pretty-printing for gdb",
                    "text": "-enable-pretty-printing",
                    "ignoreFailures": true
                }
            ],
            "preLaunchTask": "C/C++: g++ build active file",
            "miDebuggerPath": "/usr/bin/gdb",
            "logging": {
                "engineLogging": true,
                "trace": true,
                "traceResponse": true
            }
        }
    ]
}
```

tasks.json:
```json
{
    "tasks": [
        {
            "type": "shell",
            "label": "C/C++: cpp build active file",
            "command": "/usr/bin/cpp",
            "args": [
                "-g",
                "${file}",
                "-o",
                "${fileDirname}/${fileBasenameNoExtension}"
            ],
            "options": {
                "cwd": "${workspaceFolder}"
            },
            "problemMatcher": [
                "$gcc"
            ],
            "group": {
                "kind": "build",
                "isDefault": true
            }
        },
        {
            "type": "shell",
            "label": "C/C++: g++ build active file",
            "command": "/usr/bin/g++",
            "args": [
                "-g",
                "${file}",
                "-o",
                "${fileDirname}/${fileBasenameNoExtension}"
            ],
            "options": {
                "cwd": "${workspaceFolder}"
            },
            "problemMatcher": [
                "$gcc"
            ],
            "group": {
                "kind": "build",
                "isDefault": true
            }
        }
    ],
    "version": "2.0.0"
}
```

**Comment by WardenGnaw** [comment]
This is VS Code's task execution's error. It has not reached the debugger yet.

However, can you run `/usr/bin/g++ -g /home/.../test.cpp -o /home/.../test` in the commandline? 
Does it work or does it show the same error?

**Comment by allen505** [comment]
No, it does not show any error. It runs well and generates the test executable

**Comment by allen505** [comment]
@WardenGnaw , any updates on this issue or on how to fix this?

**Comment by kchoo1118** [comment]
I have the same problem on a mac.
It worked fine for clang but could not do gcc.

**Comment by zqbake** [comment]
Have the same error in linux.

**Comment by hengqujushi** [comment]
I have the same problem on a mac.

**Comment by stegra58** [comment]
I have the same problem on my MacBook Pro 2018 macOS Big Sur 11.0.1 and Visual Code 1.52.0. 
This is my Makefile:

CC := /usr/local/bin/gcc-10
F = /usr/local/opt/flex/bin/flex
B = /usr/local/opt/bison/bin/bison

fb3-2: fb3-2.l fb3-2.y fb3-2.h fb3-2funcs.c
	$(F) -o fb3-2.lex.c fb3-2.l
	$(B) -d fb3-2.y
	$(CC) -g -o $@ fb3-2.tab.c fb3-2.lex.c fb3-2funcs.c -lm

clean:
	rm -f fb3-2 fb3-2.tab.c fb3-2.tab.h fb3-2.lex.c

.PHONY: fb3-2 clean

and if I run make fb3-2 from **terminal** both in visual studio code and not,  it runs well and generates the executable **but** when i run the **task** from tasks.json I get:

collect2: fatal error: cannot find 'ld'

from tasks.json:

{
			"type": "shell",
			"label": "Build: GCC-10",
			"command": "make",
			"args": [
				 "fb3-2"
			],
			"options": {
				"cwd": "${workspaceFolder}"
			},
			"group": {
				"kind": "build",
				"isDefault": true
			}
		}

summing up:
**from terminal**: make fb3-2 **OK**
**from tasks.json**: make fb3-2 **ERROR**: collect2: fatal error: cannot find 'ld'

It should be noted that until some time ago the problem was not there and **even from tasks.json** everything worked.
Updates made: Catalina -> Big Sur, latest Visual Studio Code update.

**Comment by stegra58** [comment]
I add that if I use the intel compiler ICC everything works perfectly:

CC := /usr/local/bin/icc    # OK NO ERROR!

**Comment by yichengsu** [comment]
I also meet this problem.

But, I find I can run the task in zsh with no problem:
```
/bin/zsh -c '/usr/local/bin/g++-10 -g /path/to/main.cpp -o /path/to/main.exe'
```
which command is executed in vscode.

So, I find the difference between zsh and vscode task pannel is `$PATH`, so I use 
```
            "options": {
                "env": {
                    "PATH": "/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
                }
            }
```
to let PATH as same as my zsh terminal.

It seems to work, no errors are reported.

**Comment by Colengms** [comment]
This seems to be externa.  We don't own `shell` build tasks.  It looks like the last response provided a resolution to the issue.  Closing as external.
